src/discord/DiscordBot.ts

```diff
@@ -0,0 +1,115 @@
+import { Client, Events, GatewayIntentBits, GuildMember, Partials, REST } from "discord.js";
+import { EventHubProducerClient } from "@azure/event-hubs";
+
+import { Database, Environment } from "../utils/constants";
+import DiscordEventRouter from "./DiscordEventRouter";
+import ErrorLogger from "./ErrorLogger";
+import { MessageTelemetryLogger } from "../utils/MessageTelemetryLogger";
+import { cosmosClient } from "../db/cosmosClient";
+import InMemoryDbConnector from "../db/InMemoryDbConnector";
+import DBConnector from "../db/DBConnector";
+import { CosmosClient } from "@azure/cosmos";
+
+const FIVE_MINUTES = 1000 * 60 * 5;
+let logger: MessageTelemetryLogger | null = null;
+let eventHubMessenger;
+
+if (cosmosClient) {
+    eventHubMessenger = new EventHubProducerClient(
+        Environment.ehConnectionString,
+        Environment.Constants.telemetryEventHub
+    );
+    logger = new MessageTelemetryLogger(eventHubMessenger);
+}
+
+export default class DiscordBot {
+    private _client = new Client({
+        intents: [
+            GatewayIntentBits.Guilds,
+            GatewayIntentBits.GuildMessages,
+            GatewayIntentBits.GuildPresences,
+            GatewayIntentBits.GuildVoiceStates,
+            GatewayIntentBits.GuildMembers,
+            GatewayIntentBits.GuildMessageReactions,
+            GatewayIntentBits.GuildScheduledEvents,
+            GatewayIntentBits.MessageContent,
+        ],
+        partials: [Partials.Message, Partials.Reaction],
+    });
+    public get client() {
+        return this._client;
+    }
+
+    private _rest: REST | undefined;
+    public get rest() {
+        if (this._rest == undefined) {
+            throw new Error("You must log in before attempting to use the REST Api.");
+        }
+
+        return this._rest;
+    }
+
+    private _errorLogger;
+    public get errorLogger() {
+        return this._errorLogger;
+    }
+
+    constructor() {
+        /*
+        TODO: This code can be enabled once there is a cosmos database set up to receive errors.
+        const errorDbConnector = process.env.DEBUG
+            ? new InMemoryDbConnector<Error>()
+            : new DBConnector<Error>(cosmosClient as CosmosClient, Database.DATABASE_ID, "Errors");
+        */
+        this._errorLogger = new ErrorLogger(new InMemoryDbConnector<Error>());
+    }
+
+    public async login(botToken: string) {
+        console.log("Logging in to Discord API...");
+        await this._client.login(botToken);
+        this._rest = new REST({ version: "9" }).setToken(botToken);
```

Comment:
Wonder if we can bump this to 10 for consistency with other API methods / types

Reply:
Bumped up to 10. Appears to be working fine.